lib/Backend/CodeGenNumberAllocator.cpp

```diff
@@ -324,6 +324,9 @@ Js::JavascriptNumber* XProcNumberPageSegmentImpl::AllocateNumber(Func* func, dou
                 Recycler::FillPadNoCheck(pLocalNumber, sizeof(Js::JavascriptNumber), sizeCat, false);
                 pLocalNumber = new (pLocalNumber) Js::JavascriptNumber(localNumber);
             }
+#else
+            Assert(sizeCat == sizeof(Js::JavascriptNumber));
+            __analysis_assume(sizeCat == sizeof(Js::JavascriptNumber));
```

Comment:
There is no warning when `RECYCLER_MEMORY_VERIFY` is defined,  `pLocalNumber` is allocated as : 
`pLocalNumber = (Js::JavascriptNumber*)alloca(sizeCat);`
So at `WriteProcessMemory(hProcess, (void*)number, pLocalNumber, sizeCat, NULL)` there will be no warning from the analyzer, saying `pLocalNumber` should be >= `sizeCat`.

Also looks like when `RECYCLER_MEMORY_VERIFY` is defined `sizeCat` can be larger than `sizeof(Js::JavascriptNumber)`.